notebooks/data_processing.ipynb

```diff
@@ -0,0 +1,795 @@
+{
```

Comment:
<p>This was a good attempt with lots of improvement. I can see that meteorological values are <code>nan</code> at many places. Ideally this should not be the case. I'd suggest to follow this procedure.</p><p> </p><ul><li>For each <code>location_id</code> </li><li class="ql-indent-1">Obtain the temporal meteorological data in xarray format with <code>.sel()</code> with <code>nearest</code> keyword.</li><li class="ql-indent-1">Replace lat, lon with station's lat lon.</li><li class="ql-indent-1">Save it in a list for example, <code>met_list</code>  </li><li>Once the previous step is finished, concatenate all files within <code>met_list</code> using <code>xr.concat</code>. It should look exactly like the current format of <code>processed_data</code> but without <code>elevation</code> data and without any <code>nan</code> values.</li><li>Then get the elevation data for each <code>location_id</code> while making sure that the order is not changed. If order changes then data will become useless. You can use API at 3-4 random co-ordindates to verify that data is correct.</li><li>Now, you can just assign the elevation data to <code>met_ds["ele"] = ("location_id", ele_data)</code></li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/patel-zeel/kdd24/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
<img width="899" alt="Screenshot 2024-01-15 at 2 00 20 PM" src="https://github.com/patel-zeel/kdd24/assets/117508413/7eacb134-a6d4-4929-914b-f02e21ceaffa">

Bhaiya, I think i did exactly this only in the above lines of code. While adding elevation data i also mentioned time as a coordinate and i would change that. But i guess the era 5 data itself has missing values.